src/Microsoft.ML.Core/Data/Repository.cs

```diff
@@ -432,6 +434,24 @@ public void Commit()
             Flush();
             Dispose(true);
         }
+
+        private static string GetProductVersion()
+        {
+            var assembly = typeof(RepositoryWriter).Assembly;
+
+            var assemblyInternationalVersionAttribute = assembly.CustomAttributes.FirstOrDefault(a =>
+                    a.AttributeType == typeof(AssemblyInformationalVersionAttribute));
+
+            if (assemblyInternationalVersionAttribute == null)
+            {
+                throw new ApplicationException($"Cannot determine product version from assembly {assembly.FullName}.");
```

Comment:
>throw [](start = 16, length = 5)

Not sure we should throw in this case. Perhaps we should save `0.0.0.0` in this case? Or, alternatively, not include `Version.txt` in the model?

@eerhardt, @harishsk

Reply:
This was only to keep consistency with how it was behaving before - I prefer the approach you have suggested though it should not fail because of this.

Reply:
Please confirm @eerhardt @harishsk @yaeldekel  and I'll make the change.